File: ts/256/205.ts
// 哈希表
function isIsomorphic(s: string, t: string): boolean {
  const sht = {}, tht = {}
  for (let i = 0, len = s.length; i < len; ++i) {
    if ((sht[s[i]] && sht[s[i]] !== t[i]) ||
      (tht[t[i]] && tht[t[i]] !== s[i])
    ) return false
    sht[s[i]] = t[i]
    tht[t[i]] = s[i]
  }
  return true
};

File: py3/256/205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        n = len(s)
        sht, tht = {}, {}
        for i in range(n):
            if (s[i] in sht and sht[s[i]] != t[i]) or (t[i] in tht and tht[t[i]] != s[i]):
                return False
            sht[s[i]] = t[i]
            tht[t[i]] = s[i]

        return True